### PR TITLE
Add or complete ddoc style comments to all target functions and fields

### PR DESCRIPTION
**Pull request by ibuclaw**

### CODE REVIEW

File: src/dmd/target.d [hunk 1]
```diff
@@ -26,50 +26,62 @@ import dmd.tokens : TOK;
...
+            real_t max = T.max;                 /// largest representable value that's not infinity
+            real_t min_normal = T.min_normal;   /// smallest representable normalized value that's not 0
+            real_t nan = T.nan;                 /// NaN value
+            real_t snan = T.init;               /// signalling NaN value
+            real_t infinity = T.infinity;       /// infinity value
+            real_t epsilon = T.epsilon;         /// smallest increment to the value 1
+
+            d_int64 dig = T.dig;                /// number of decimal digits of precision
+            d_int64 mant_dig = T.mant_dig;      /// number of bits in mantissa
+            d_int64 max_exp = T.max_exp;        /// maximum int value such that 2$(SUPERSCIPT `max_exp-1`) is representable
```
Comment (ibuclaw) [inline src/dmd/target.d]: Excellent, I'm using undefined features.

File: src/dmd/target.d [hunk 2]
```diff
@@ -26,50 +26,62 @@ import dmd.tokens : TOK;
 import dmd.root.ctfloat;
 import dmd.root.outbuffer;
 
-/***********************************************************
+/**
+ * A data structure that describes a back-end target. At present it is
```
Comment (WalterBright) [inline src/dmd/target.d]: s/A data structure that describes/Describes/

I'm a fan of the notion that if words can be removed from a sentence with no loss of meaning, the words should be excised.

File: src/dmd/target.d [hunk 3]
```diff
@@ -146,8 +158,10 @@ struct Target
         uint64Mangle = global.params.isOSX ? 'y' : 'm';
     }
 
-    /******************************
-     * Return memory alignment size of type.
+    /**
+     * Return memory alignment size of the given type.
```
Comment (WalterBright) [inline src/dmd/target.d]: Need:
```
Returns:
```
Ddoc section

Comment (WalterBright) [reply src/dmd/target.d]: Above comment applies globally to this file.

File: src/dmd/target.d [hunk 4]
```diff
@@ -191,8 +207,8 @@ struct Target
         return (8 < size) ? 8 : size;
     }
 
-    /***********************************
-     * Return size of OS critical section.
+    /**
+     * Returns the size of the target OS critical section.
      * NOTE: can't use the sizeof() calls directly since cross compiling is
      * supported and would end up using the host sizes rather than the target
      * sizes.
```
Comment (WalterBright) [inline src/dmd/target.d]: This note would seem to apply globally to every symbol in this file.

File: src/dmd/target.d [hunk 5]
```diff
@@ -383,10 +403,13 @@ struct Target
...
 
-    /******************************
+    /**
      * Encode the given expression, which is assumed to be an rvalue literal
      * as another type for use in CTFE.
-     * This corresponds roughly to the idiom *(Type *)&e.
+     * This corresponds roughly to the idiom `*cast(T*)&e`.
+     * Params:
+     *      e    = literal constant expression
+     *      type = target type of the result
```
Comment (WalterBright) [inline src/dmd/target.d]: Missing `Returns`

**Comment by ibuclaw** [comment]
Added returns, re-worded the original comments in some places so there aren't redundant comments, i.e:
```
Return the field alignment in bytes of the given type.
Returns:
    The field alignment in bytes of the given type.
```

**Comment by ibuclaw** [comment]
@WalterBright - Ping?

File: src/dmd/target.d [hunk 6]
```diff
@@ -26,50 +26,66 @@ import dmd.tokens : TOK;
 import dmd.root.ctfloat;
 import dmd.root.outbuffer;
 
-/***********************************************************
+/**
+ * Describes a back-end target. At present it is incomplete, but in future it
```
Comment (JinShil) [inline src/dmd/target.d]: in *the* future

File: src/dmd/target.d [hunk 7]
```diff
@@ -26,50 +26,66 @@ import dmd.tokens : TOK;
...
+ * should grow to contain most or all target machine and target O/S specific
+ * information.
+ *
+ * In many cases, can't use the sizeof() calls directly for getting data type
```
Comment (JinShil) [inline src/dmd/target.d]: Suggested wordsmithing: In many cases, calls to \`sizeof()\` can't be used...

File: src/dmd/target.d [hunk 8]
```diff
@@ -479,25 +533,35 @@ struct Target
...
      */
     extern (C++) static LINK systemLinkage()
     {
         return global.params.isWindows ? LINK.windows : LINK.c;
     }
 
     /**
-     * Return a tuple describing how argument type is put to a function.
-     * Value is an empty tuple if type is always passed on the stack.
+     * Describes how an argument type is put to a function on target.
```
Comment (JinShil) [inline src/dmd/target.d]: "passed to" instead of "put to"?

**Comment by ibuclaw** [comment]
Fixed up and rebased.

**Review by JinShil — APPROVED** [review]

**Review by jacob-carlborg — APPROVED** [review]

**Pull request merged**